examples/10_streaming_read.cpp

```diff
@@ -0,0 +1,63 @@
+#define BUILD_STREAMING_EXAMPLE false
+#if BUILD_STREAMING_EXAMPLE
```

Comment:
Our CI will automatically run all the examples, one after another. But the streaming examples need to be run concurrently, so the easiest solution for now was to just deactivate them by default. Otherwise, the CI will just run into a timeout, waiting for the reader/writer indefinitely.

Reply:
Discussed today: automatic coverage will be enabled in a follow-up PR, so things don't break over time with different contributors to the code base.

Idea: Single MPI context and split for a test.